core/src/banking_stage/consume_worker.rs

```diff
@@ -775,10 +993,18 @@ pub(crate) mod external {
         }
 
         fn validate_message_flags(flags: u16) -> bool {
-            flags == pack_message_flags::EXECUTE
-                || (flags
-                    == (pack_message_flags::CHECK
-                        | pack_message_flags::check_flags::LOAD_ADDRESS_LOOKUP_TABLES))
+            if flags & pack_message_flags::EXECUTE != 0 {
+                const ALLOWED_EXECUTE_FLAGS: u16 = pack_message_flags::EXECUTE;
+
+                flags & !ALLOWED_EXECUTE_FLAGS == 0
+            } else {
+                const ALLOWED_CHECK_BITS: u16 = pack_message_flags::CHECK
+                    | check_flags::STATUS_CHECKS
+                    | check_flags::LOAD_FEE_PAYER_BALANCE
+                    | check_flags::LOAD_ADDRESS_LOOKUP_TABLES;
+
+                flags != pack_message_flags::CHECK && flags & !ALLOWED_CHECK_BITS == 0
```

Comment:
Is this just confirming that at least one of status checks, fee payer, alt is set (and nothing else)?

Reply:
Yeah, just considered invalid without any checks requested.